cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -139,14 +139,13 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
                                "in project '%s' is in the ACTIVE state. Prefer External IP can be set to false to " +
                                "launch Dataproc clusters with internal IP only.", systemNetwork,
                              systemProjectId, network, projectId));
-    } else if (state == PeeringState.INACTIVE && !conf.isPreferExternalIP()) {
-      // Peering is setup between the system network and customer network. However it is not in ACTIVE state.
-      // User has also intended to use the internal IP. Dataproc cluster cannot be launched on the internal IP unless
-      // the peering is fixed.
+    } else if (!conf.isPreferExternalIP() && state != PeeringState.ACTIVE) {
```

Comment:
this was correct before and incorrect now. "Prefer External IP" just means always use the external ip if it is true. If it is false, let the provisioner decide. A false value does *not* mean always use the internal ip.